### DISCUSSION TITLE
How to use SkiaSharp with Uno 4.7?

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#11080] [Q&A] [answered]
Maybe I'm doing something wrong - but what is the correct way to use SkiaSharp within e.g. the MainPage.Xaml of a newly created Uno-App?

So:
1. Create a new app
2. Add Nuget "Skiasharp" to Shared-Class-library
3. Open MainPage.xaml and add a SkXamlCanvas.

I'm lost on Step 3 - where do I find the SkXamlCanvas? I assume it is in a "views"-nuget from Skiasharp and I would expect that it is "SkiaSharp.Views.Uno". But I did not get the app to compile as I could not find the correct Namespace for the Xaml.
It get's even more confusing as there is also a "SkiaSharp.Views"-nuget and a "SkiaSharp.Views.Uno.WinUi" IIRC.

So the question is: what is the latest way to get something drawn on a Canvas with Skia on all heads. :) Thank you!

**Comment by jeromelaban** [comment, marked as answer]
We're in the process of updating our sample about SkiaSharp in https://github.com/unoplatform/Uno.Samples/pull/270, but you can browse the code for [that sample here](https://github.com/unoplatform/Uno.Samples/tree/dev/AEST/190-update-to-net6-skia/UI/SkiaSharpTest).

**Reply by TopperDEL** [reply]
Thanks! But the sample URL is not valid as it seems.

**Reply by mcNets** [reply]
Try [this](https://github.com/unoplatform/Uno.Samples/tree/master/UI/SkiaSharpTest) one

**Reply by jeromelaban** [reply]
It got merged before you visited it :) @mcNets's version is the official now.

**Reply by TopperDEL** [reply]
Ah, that's a good reason. 🙈👍

**Reply by TopperDEL** [reply]
Hm, sorry @jeromelaban but that does not help with the new class-library-template. With a shared-project I just add the references to the heads that fit the head. But here I would need a "generic" SkiaSharp.Views-Nuget for the class library to get the SkXamlCanvas-Object in my MainPage.xaml. But which one to use then?

**Reply by jeromelaban** [reply]
Indeed, the sample is not updated to the latest templates, which we'll do soon.

In the meantime, in the separate library, you'll need to adjust references based on the platform:

```xml
<ItemGroup Condition="'$(TargetFramework)' == 'net6.0-windows10.0.18362.0'">
  <PackageReference Include="SkiaSharp.Views.WinUI" Version="2.88.3" />
</ItemGroup>
<ItemGroup Condition="'$(TargetFramework)' != 'net6.0-windows10.0.18362.0'">
  <PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.3" />
</ItemGroup>
```

You'll need to adjust the condition to use the targetframework used in your project.

**Reply by TopperDEL** [reply]
Got it, thanks!
